python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+        for key, val in row.items():
+            if val == "":
+                row[key] = None
+    return rows
+
+
+def handle_crash_date_time(rows):
+    """Combine crash_date and crash_time and format as ISO string.
+
+    The same ISO timestamp is stored as `crash_date` and `crash_time`
+    """
+    tz = pytz.timezone("US/Central")
+    for row in rows:
+        # parse a string that looks like '12/22/2023'
+        crash_date = row["crash_date"]
+        month, day, year = crash_date.split("/")
+        # parse a string that looks like '12:19 PM'
+        crash_time = row["crash_time"]
+        hour_minute, am_pm = crash_time.split(" ")
+        hour, minute = hour_minute.split(":")
+        if am_pm.lower() == "pm":
+            hour = int(hour) + 12 if hour != "12" else int(hour)
+        else:
+            hour = int(hour) if hour != "12" else 0
+        # create a tz-aware instance of this datetime
+        crash_date_dt = tz.localize(
+            datetime(int(year), int(month), int(day), hour=hour, minute=int(minute))
+        )
+        # save the ISO string with tz offset
+        crash_date_iso = crash_date_dt.isoformat()
+        row["crash_date"] = crash_date_iso
+        row["crash_time"] = crash_date_iso
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def main():
+    files_todo = get_files_todo()
+    extract_ids = list(set([f["extract_id"] for f in files_todo]))
+    # assumes extract ids are sortable oldest > newest by filename. todo: is that right?
+    extract_ids.sort()
+    for extract_id in extract_ids:
+        # get schema years that match this extract ID
+        schema_years = list(
+            set([f["schema_year"] for f in files_todo if f["extract_id"] == extract_id])
+        )
+        schema_years.sort()
+        for schema_year in schema_years:
+            for table_name in ["crashes", "units", "persons"]:
+                file = next(
+                    (
+                        f
+                        for f in files_todo
+                        if f["extract_id"] == extract_id
+                        and f["schema_year"] == schema_year
+                        and table_name.startswith(f["table_name"])
+                    ),
+                    None,
+                )
+
+                if not file:
+                    raise (
+                        f"No {table_name} file found in extract. This should never happen!"
+                    )
+
+                print(f"processing {table_name}")
+
+                records = handle_empty_strings(
+                    remove_unsupported_columns(
+                        lower_case_keys(load_csv(file["path"])), table_name
+                    )
+                )
+
+                if table_name == "crashes":
+                    handle_crash_date_time(records)
+
+                # annoying redundant branch to combine primary persons and persons
```

Comment:
i think we'll be able to tighten up some of this branching logic, and the annoying `next(...)` calls.